src/server/api/url_parser.cpp

```diff
@@ -26,15 +26,15 @@ struct URLParser final : qi::grammar<Iterator, Into>
     {
         using boost::spirit::repository::qi::iter_pos;
 
-        alpha_numeral = qi::char_("a-zA-Z0-9");
+        identifier = qi::char_("a-zA-Z0-9_..--~:");
```

Comment:
Yes, moving the `-` to the end seems to prevent it from forming a range, and removes the need for `..`
```cpp
identifier = qi::char_("a-zA-Z0-9_.~:-");
```